tests/components/obihai/test_config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Test the Obihai config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.obihai.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import USER_INPUT
+
+
+async def test_user_form(hass: HomeAssistant) -> None:
+    """Test we get the user initiated form."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+
+    with patch("pyobihai.PyObihai.check_account"), patch(
+        "homeassistant.components.obihai.async_setup_entry"
+    ) as mock_setup_entry:
```

Comment:
Please take a look at https://github.com/home-assistant/core/pull/88661
I think it's better to ensure `async_setup_entry` is always mocked in config flow tests (with `pytestmark = pytest.mark.usefixtures`), and then inject the fixture in the test definition.